### PR TITLE
Prohibit positional bool

### LINKED ISSUE #212: Weird bool behaviour when not defined as a flag parameter
I've been (incorrectly) using `bool` arguments without any option under the assumption that the mandatory value is actually parsed and turned into a proper `bool` according to what was passed on the CLI but I just found out that it is always `true`. This makes `bool` parameters which are not flags rather useless; IMHO they either need to be really parsed or there should be a compiler error saying: `You've been trying to use a bool parameter which is not a flag and hence will be always true after evaluation which is probably not what you want`.

### PR DESCRIPTION
**Pull request by CreepySkeleton**
Closes #212

cc @TeXitoi please also take a look at #292

### CODE REVIEW

**Comment by therealprof** [comment]
The example is good, the "documentation" does not help a bit.

**Comment by CreepySkeleton** [comment]
@therealprof This to address #287 , your case is covered by the example.

**Comment by therealprof** [comment]
Well, it didn't say that on the label, maybe you should throw in a "Closes #287" then?

**Comment by CreepySkeleton** [comment]
@therealprof 
#287 is not an issue but PR, take look at the first comment there for detailed information.
I included it in this PR because "see this for example" link refers to `examples/true_or_false.rs` which is to be added by this PR.

**Comment by CreepySkeleton** [comment]
> The example is good, the "documentation" does not help a bit.

Also, @therealprof , if you have some idea about how to address your case in the doc I'm all ears.

**Comment by CreepySkeleton** [comment]
@therealprof 

I would like to apologize for being too blunt. I'm not really a "public relations" man, I take strait approach when explaining thinks to others, too strait sometimes. It was never my intention to offend you or discard your opinion, sorry if I left wrong impression.

But I'm not making amends, I mean every word I said. I still believe you case is a corner one, `true/false` arguments are common for config files indeed, but I have yet to meet a case when I would need to actually use them on command line. Even more, I wasn't even aware of existence of such applications until after you'd brought some examples. 

We can't bring every corner case to README (and it's not an actual documentation anyway), hence I'm closing your PR. Do you think this PR provides adequate explanation?

File: examples/true_or_false.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+//! How to parse `--foo=true --bar=false` and turn them into bool.
+
+use structopt::StructOpt;
+
+fn true_or_false(s: &str) -> Result<bool, &'static str> {
+    match s {
+        "true" => Ok(true),
+        "false" => Ok(false),
```
Comment (TeXitoi) [inline examples/true_or_false.rs]: Maybe "yes" or "no" would be a more meaningful example.

File: examples/true_or_false.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    // Default parser for `try_from_str` is FromStr::from_str.
+    // `impl FromStr for bool` parses `true` or `false` so this
+    // works as expected.
+    #[structopt(long, parse(try_from_str))]
+    foo: bool,
+
+    // Of course, this could be done with an explicit parser function.
+    #[structopt(long, parse(try_from_str = true_or_false))]
+    bar: bool,
+}
```
Comment (TeXitoi) [inline examples/true_or_false.rs]: It could be great to also add a boolean parameter for completeness.

File: src/lib.rs [hunk 3]
```diff
@@ -302,6 +302,22 @@
...
+//! Pay attention that *only literal occurrence* of this types is special, for example
+//! `Option<T>` is special while `::std::option::Option<T>` is not.
+//!
+//! If you need to avoid special casing you can make a `type` alias and
+//! use it in place of the said type.
+//! _________________
+//!
+//! **Note:**
+//! _________________
+//! `bool` cannot be used as positional argument with an implicit parser. If you need
```
Comment (TeXitoi) [inline src/lib.rs]: I'd prefer "without an explicit parser"

File: src/lib.rs [hunk 4]
```diff
@@ -302,6 +302,22 @@
...
+//! a positional `bool`, for example to parse `true` or `false` add an explicit
+//! [`#[structopt(parse(...))]` annotation](#custom-string-parsers).
```
Comment (TeXitoi) [inline src/lib.rs]: If you need a positional `bool`, for example to parse `true` or `false`, you can add an explicit [`#[structopt(parse(try_from_str))]` annotation](#custom-string-parsers).

File: examples/true_or_false.rs (hunk 2 above)
Comment (CreepySkeleton) [reply examples/true_or_false.rs]: Do you mean positional parameter?

File: examples/true_or_false.rs (hunk 1 above)
Comment (CreepySkeleton) [reply examples/true_or_false.rs]: This example is meant to demonstrate how `impl FromStr for bool` works, the intention is to show how to do the same thing in two different ways.

File: examples/true_or_false.rs (hunk 2 above)
Comment (TeXitoi) [reply examples/true_or_false.rs]: yep

File: examples/true_or_false.rs (hunk 1 above)
Comment (TeXitoi) [reply examples/true_or_false.rs]: OK

**Comment by therealprof** [comment]
@CreepySkeleton No worries, all good here.

File: examples/true_or_false.rs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        Opt::from_iter(&["test", "--foo=true", "--bar=false", "true"]),
+        Opt {
+            foo: true,
+            bar: false,
+            boom: true
+        }
+    );
+    // no beauty, only truth and falseness
+    assert!(Opt::from_iter_safe(&["test", "--foo=beauty"]).is_err());
+}
```
Comment (TeXitoi) [inline examples/true_or_false.rs]: Examples are normally working programs that output the Opt struct to stdout, not unit test like.

Comment (CreepySkeleton) [reply examples/true_or_false.rs]: Yes, I just wanted to show "what's the result is" in the most demonstrative way.

**Pull request merged**